app/scripts/modules/core/src/managed/Pill.module.css

```diff
@@ -9,3 +9,19 @@
   -moz-osx-font-smoothing: grayscale;
   text-rendering: optimizeLegibility;
 }
+
+.opacityAnimationEnabled {
+  animation-direction: alternate;
+  animation-duration: 500ms;
+  animation-iteration-count: infinite;
+  animation-name: animate-opacity;
+}
+
+@keyframes animate-opacity {
+  from {
+    opacity: 0.5;
+  }
+  to {
+    opacity: 1;
```

Comment:
because this is an opacity adjustment across the entire subtree of the pill, am I correct in thinking the text also fades in/out? If so, can you think of a good way to have the background pulse here without fading the text?

Reply:
> because this is an opacity adjustment across the entire subtree of the pill, am I correct in thinking the text also fades in/out?

That is correct.

> If so, can you think of a good way to have the background pulse here without fading the text?

I might have to use JS based animation since the background opacity animation may not be possible with css animations and the background color are dynamically provided as props. Let me try the JS animation approach.

Reply:
cc @gcomstock in case he has opinions or suggestions from having done this elsewhere — one other way is to animate the background color between two points vs. animating opacity.

Reply:
another thing to consider: maybe you could just have the background be it's own sibling element alongside the text (i.e. behind it) instead of being applied to the containing element.

Reply:
I ended up using `react-spring` to perform the animation and now only the [background color's opacity](https://github.com/spinnaker/deck/pull/8123/files#diff-8f60187ca9e9849993494ad0afd80689R71) is animated.

Reply:
@gcomstock since we have js based animation available, we can potentially change this behavior to any form of animation you prefer. [This](https://codesandbox.io/s/quiet-framework-z4x0p) is the current behavior.

Reply:
Looks fun! I think gentle is fine. We can change our minds if needed as we introduce more animations elsewhere

Reply:
+1 to gentle, might also be worth experimenting with slowing it down by like 15% if that's easy to do (i didn't see a way in the codesandbox?)

btw: that sandbox is 🔥 🔥 🔥

Reply:
Updated to `gentle` now.

@erikmunson I've updated codesandbox to accept custom values. You can play with `tension` and `friction` values to alter the animation. The values for the presets can be found [here](https://www.react-spring.io/docs/hooks/api)

Reply:
wowowow this is so cool thanks for setting it up! I am really excited to start using these physics rules on other transitions 💥 

If you're open to it, I think it'd be nice to add some mass and overall sluggishness to the pulse — I personally gravitate toward pulsing animations that feel almost like a calm (but not too slow) breath in and out. After playing with that config maybe something like:
```
{
  mass: 16,
  tension: 390,
  friction: 130
}
```

It'd be really magic if down the road we could get a standard, well-liked "pulse" spring and re-use that across anything that needs to do some kind of animation like this.

Reply:
I like this behavior, updating it!